pyomo/contrib/mindtpy/tests/test_mindtpy_lp_nlp.py

```diff
@@ -23,12 +23,18 @@
 from pyomo.environ import SolverFactory, value
 from pyomo.opt import TerminationCondition
 
-required_solvers = ('ipopt', 'cplex_persistent')
-if all(SolverFactory(s).available(False) for s in required_solvers):
+required_nlp_solvers = 'ipopt'
+required_mip_solvers = ['cplex_persistent', 'gurobi_persistent']
+
+
+if SolverFactory(required_nlp_solvers).available(False) and any(SolverFactory(s).available(False) for s in required_mip_solvers):
     subsolvers_available = True
+    available_mip_solvers = [s for s in required_mip_solvers
+                             if SolverFactory(s).available(False)]
 else:
     subsolvers_available = False
```

Comment:
My read of this is that the single tree tests use either cplex or gurobi - whichever happens to be present.  Can you consider having (a subset of) these tests duplicated and tied explicitly to Cplex and Gurobi?

Reply:
No problem.